pkg/influx/api.go

```diff
@@ -93,22 +68,19 @@ func (a *API) handleSeriesPush(w http.ResponseWriter, r *http.Request) {
 			TimeSeries: &ts[i],
 		})
 	}
-
 	rwReq := &cortexpb.WriteRequest{
 		Timeseries: pts,
 	}
 
-	if _, err := a.client.Push(r.Context(), rwReq); err != nil {
-		resp, ok := httpgrpc.HTTPResponseFromError(err)
-		if !ok {
-			level.Warn(a.logger).Log("msg", "failed to push metric data", "err", err)
-			http.Error(w, err.Error(), http.StatusInternalServerError)
+	if err := a.client.Write(r.Context(), rwReq); err != nil {
+		if errors.As(err, &errorx.RateLimited{}) {
```

Comment:
I don't think we need to check for ratelimited specifically, since ratelimited has a useful error message already.  Instead, do (pseudocode):

```
if errors.As(err, &errorx.Error) {
  if HTTPStatusCode() is a 400 {
     log info
   } else if code is a 500 {
     log warning
   }
   http.Error(   // using the values from the Error)
   return
}
// else...
non-errorx, should not happen.  log warning and set InternalError
```

Reply:
this block of error handling might even be worthy of putting in errorx itself and passing in the response, since it is generic

Reply:
I have another PR with much more detailed error handling, so this will be improved in that PR :)